strongbox-cron-tasks/src/main/java/org.carlspring.strongbox/cron/api/jobs/RebuildMetadataCronJob.java

```diff
@@ -0,0 +1,171 @@
+package org.carlspring.strongbox.cron.api.jobs;
+
+import org.carlspring.strongbox.cron.config.ApplicationContextProvider;
+import org.carlspring.strongbox.cron.config.JobManager;
+import org.carlspring.strongbox.cron.domain.CronTaskConfiguration;
+import org.carlspring.strongbox.resource.ConfigurationResourceResolver;
+import org.carlspring.strongbox.services.impl.ArtifactMetadataServiceImpl;
+
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Kate Novik
+ */
+public class RebuildMetadataCronJob
+        extends JavaCronJob
+{
+
+    private static final String BASEDIR_STORAGES = "storages";
+
+    private static final String storagesBasePath = getStoragesBasePath();
+
+    private final Logger logger = LoggerFactory.getLogger(RebuildMetadataCronJob.class);
+
+    private ArtifactMetadataServiceImpl artifactMetadataService = getArtifactMetadataService();
```

Comment:
@kate-novik , @AlexOreshkevich : 

- This needs to be properly `@Autowired`.
- It should also be using the interface `private ArtifactMetadataService artifactMetadataService;` and not the implementations `ArtifactMetadataServiceImpl`.